os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_flash_ram.c

```diff
@@ -86,15 +86,15 @@ void FLASH_Write_Lock(void)
 #endif
 	/* disable irq */
 	/* We do not need to acquire lock for this core, as the other core will enter gating */
-	PrevIrqStatus = irqsave();
+	PrevIrqStatus = enter_critical_section();
```

Comment:
Thanks for identifying this, the irqsave/restore was set here purposely during previous work. I found some booting hang issue and think that the core doing flash operation should not compete for spin lock.
May I know if this change is able to cover all the cases, I mean under complex situation? (With considering the change in irq_csection.c)